### PR TITLE
fix issue that pipeline-service setup task take the correct revision

### PR DESCRIPTION
**Pull request by xinredhat**

### CODE REVIEW

**Review by ramessesii2 — APPROVED** [review]

**Comment by gabemontero** [comment]
CI is green @Roming22 @xinredhat 

should we convert this from draft to reviewable PR and get this merged?

**Comment by Roming22** [comment]
I don't think this PR should be merged. Have a look at #761, and you will see that we still would not have the expected behavior.

#760 might be a better fix.

File: .tekton/pipeline/upgrade-tests.yaml [hunk 1]
```diff
@@ -90,7 +90,7 @@ spec:
         - name: repo_url
           value: $(params.repo_url)
         - name: revision
-          value: $(params.revision)
+          value: $(params.target_branch)
```
Comment (Roming22) [inline .tekton/pipeline/upgrade-tests.yaml:93]: I don't think that is the right fix. If changes are done to the script in the `ci` folder, they won't be tested unless the version checked out is the `revision`.

I believe the proper fix is to checkout the right branch before every test.

**Review by Roming22 — CHANGES_REQUESTED** [review]

File: .tekton/pipeline/upgrade-tests.yaml (hunk 1 above)
Comment (xinredhat) [reply .tekton/pipeline/upgrade-tests.yaml:93]: the each git-clone task does checkout the right branch before every test, so the `revision` should align with the git-clone task, for example https://github.com/openshift-pipelines/pipeline-service/blob/main/.tekton/pipeline/upgrade-tests.yaml#L66

File: .tekton/pipeline/acceptance-tests.yaml [hunk 2]
```diff
@@ -52,17 +63,6 @@ spec:
           workspace: shared-workspace
         - name: source
           workspace: source
-    - name: clone-pipeline-service-git
```
Comment (Roming22) [inline .tekton/pipeline/acceptance-tests.yaml:55]: Not sure why this change is necessary.

File: .tekton/tasks/ci-runner-container-setup.yaml [hunk 3]
```diff
@@ -22,15 +22,15 @@ spec:
...
-        requests:
-          memory: 500Mi
-          cpu: 300m
-      workingDir: "$(workspaces.source.path)"
-      env:
-        - name: KUBECONFIG
-          value: "$(workspaces.kubeconfig-dir.path)/kubeconfig"
-      command:
-        - $(workspaces.source.path)/ci/images/ci-runner/hack/bin/copy-plnsvc-code.sh
+    # - name: copy-plnsvc-code
```
Comment (Roming22) [inline .tekton/tasks/ci-runner-container-setup.yaml:25]: Delete.

File: ci/images/ci-runner/hack/bin/copy-plnsvc-code.sh [hunk 4]
```diff
@@ -5,7 +5,9 @@ set -o pipefail
 set -x
 
 echo "Copy source code of pipeline service to the ci-runner container"
+kubectl exec -n default -it ci-runner -- bash -c "rm -rf /source/*"
```
Comment (Roming22) [inline ci/images/ci-runner/hack/bin/copy-plnsvc-code.sh:8]: I'm expecting this to fail on the very first run, thus ending the PLR in failure. Check for the folder existence before deletion.

Same for the next line.

**Review by Roming22 — CHANGES_REQUESTED** [review]
That's better but still likely to fail. 

I haven't wrapped my head around it yet, but I am wondering if the content of `ci/images/ci-runner/hack/sidecar` needs to always point to the PR content on the ci-runner container.

File: .tekton/pipeline/acceptance-tests.yaml (hunk 2 above)
Comment (xinredhat) [reply .tekton/pipeline/acceptance-tests.yaml:55]: Yes, I just moved the block to the front. Because other tasks run after it. this movement is  helpful to understand the whole pipeline logic.

File: .tekton/tasks/ci-runner-container-setup.yaml (hunk 3 above)
Comment (xinredhat) [reply .tekton/tasks/ci-runner-container-setup.yaml:25]: done.

**Comment by xinredhat** [comment]
> but I am wondering if the content of `ci/images/ci-runner/hack/sidecar` needs to always point to the PR content on the ci-runner container.

I don't quite understand why you are considering it. the content of `ci/images/ci-runner/hack/sidecar` is part of the repo, it depends on which branch the git-clone task check out.

File: ci/images/ci-runner/hack/bin/copy-plnsvc-code.sh (hunk 4 above)
Comment (xinredhat) [reply ci/images/ci-runner/hack/bin/copy-plnsvc-code.sh:8]: I am confusing why you are expecting this to fail on the very first run, because directory `/source` already is there , it always succeed no matter if there are files or not under  `/source/`

**Comment by Roming22** [comment]
@xinredhat  Can we close this PR since #760 has solved the issue?
